packages/editor/src/lib/editor/Editor.ts

```diff
@@ -1598,11 +1598,22 @@ export class Editor extends EventEmitter<TLEventMap> {
 		return this
 	}
 
+	/**
+	 * The id of the app's only selected shape.
+	 *
+	 * @returns Null if there is no shape or more than one selected shape, otherwise the selected shape's id.
+	 *
+	 * @public
+	 * @readonly
+	 */
+	@computed getOnlySelectedShapeId(): TLShapeId | null {
+		return this.getOnlySelectedShape()?.id ?? null
+	}
+
```

Comment:
There are a lot of places in the app where we check whether a shape is the selected shape. Using the regular `getOnlySelectedShape` works but that computed fires every time the selected shape changes, which means all of those components re-render when the selected shape changes, ie when you're typing in a label.

Reply:
This computed only tells us _which_ shape is editing, which is useful for "am i selected" checks.

Reply:
killer - makes sense!